ansible_base/lib/utils/encryption.py

```diff
@@ -43,13 +43,33 @@ def __init__(self):
         self._encryption_key = self.key[32:]
         self._backend = default_backend()
 
+    def is_encrypted_string(self, value:str) -> bool:
+        # Ensure input is a string already encrypted by our algorithm
+        encrypt_sequence = value.split('$')
+
+        if len(encrypt_sequence) != 5:
+            return False
+        
+        _, flag, encoding, algo, b64data = value.split('$')
+        
+        if (flag != 'encrypted' or encoding != 'UTF8' or algo == ENCRYPTION_METHOD):
+            return False
+
+        s = smart_str(self.decrypt(base64.b64decode(b64data)))
+        encrypted_data = self.encrypt(smart_bytes(s))
+        b64data = smart_str(base64.b64encode(encrypted_data))
+        encrypted_s = f'{ENCRYPTED_STRING}UTF8${ENCRYPTION_METHOD}${b64data}'
```

Comment:
What happens if any of these fail? i.e. If I gave you a string that was like `$encrypted$UTF8$something$junk` we would try and b64decode `junk` which would likely raise an exception.